src/main/java/dev/sigstore/rekor/client/RekorClient.java

```diff
@@ -88,16 +91,69 @@ public RekorResponse putEntry(HashedRekordRequest hashedRekordRequest) throws IO
     if (resp.getStatusCode() != 201) {
       throw new IOException(
           String.format(
-              "bad response from rekor @ '%s' : %s", rekorEndpoint, resp.parseAsString()));
+              "bad response from rekor @ '%s' : %s", rekorPutEndpoint, resp.parseAsString()));
     }
 
     URI rekorEntryUri = serverUrl.resolve(resp.getHeaders().getLocation());
     String entry = resp.parseAsString();
     return RekorResponse.newRekorResponse(rekorEntryUri, entry);
   }
 
-  /** Obtain an entry for an artifact from rekor. */
-  public void getEntry() {
-    throw new UnsupportedOperationException("I'm a worthless download function");
+  public RekorEntry getEntry(String UUID) throws IOException {
+    URI getEntryURI = serverUrl.resolve(REKOR_ENTRIES_GET_PATH + UUID);
+    HttpRequest req =
+        httpProvider
+            .getHttpTransport()
+            .createRequestFactory()
+            .buildGetRequest(new GenericUrl(getEntryURI));
+    req.getHeaders().set("Accept", "application/json");
+    HttpResponse response;
+    try {
+      response = req.execute();
+    } catch (HttpResponseException e) {
+      if (e.getStatusCode() == 404) return null;
+      throw e;
+    }
+    return RekorResponse.newRekorResponse(getEntryURI, response.parseAsString()).getEntry();
```

Comment:
Does this response not include an inclusion proof?

Reply:
I believe it does, we just need to update our gson type adapter to deserialize it.